micro-rdk/src/common/registry.rs

```diff
@@ -133,6 +137,27 @@ type GenericComponentConstructor =
 
 type DependenciesFromConfig = dyn Fn(ConfigType) -> Vec<ResourceKey>;
 
+/// Macro for modules to create a registration key for their component models.
+/// The first argument should be the name of the organization responsible for
+/// creating and maintaining the module and the second argument should be the name
+/// of the module
+///
+/// Example
+///
+/// ```ignore
+/// pub fn register_models(registry: &mut ComponentRegistry) -> Result<(), RegistryError> {
+///     registry.register_motor(model!("viam", "water_pump"), &WaterPump::from_config)?;
+///     log::info!("water_pump motor registration ok");
+///     Ok(())
+/// }
+/// ```
+#[macro_export]
+macro_rules! model {
+    ($org:tt, $mname:tt) => {
+        concat!($org, ":", env!("CARGO_PKG_NAME"), ":", $mname)
+    };
+}
+
```

Comment:
can we move this to the `micro-rdk-macros` crate?

Reply:
I'd rather not The only reason that crate exists is because proc-macros currently need to be in their own crate